app/src/main/java/com/nervousfish/nervousfish/activities/CreateProfileActivity.java

```diff
@@ -0,0 +1,165 @@
+            final String name = nameInputField.getText().toString();
+
+            final KeyPair keyPair = this.generateKeyPair();
+
+            try {
+                serviceLocator.getDatabase().addProfile(new Profile(name, keyPair));
+                new SweetAlertDialog(this, SweetAlertDialog.SUCCESS_TYPE)
+                        .setTitleText(getString(R.string.profile_created))
+                        .setContentText(getString(R.string.profile_created_explanation))
+                        .setConfirmText(getString(R.string.dialog_ok))
+                        .setConfirmClickListener(new SweetAlertDialog.OnSweetClickListener() {
+                            @Override
+                            public void onClick(final SweetAlertDialog sDialog) {
+                                sDialog.dismiss();
+                                final Intent intent = new Intent(CreateProfileActivity.this, LoginActivity.class);
+                                intent.putExtra(ConstantKeywords.SERVICE_LOCATOR, serviceLocator);
+                                CreateProfileActivity.this.startActivity(intent);
+                            }
+                        })
+                        .show();
+            } catch (IOException e) {
+                new SweetAlertDialog(this, SweetAlertDialog.WARNING_TYPE)
+                        .setTitleText(getString(R.string.could_not_create_profile))
+                        .setContentText(getString(R.string.could_not_create_profile_explanation))
+                        .setConfirmText(getString(R.string.dialog_ok))
+                        .show();
+            }
+        } else {
+            new SweetAlertDialog(this, SweetAlertDialog.WARNING_TYPE)
+                    .setTitleText(getString(R.string.could_not_create_profile))
+                    .setContentText(getString(R.string.could_not_create_profile_explanation))
+                    .setConfirmText(getString(R.string.dialog_ok))
+                    .show();
+        }
+    }
+
+    /**
+     * Generates a KeyPair based on the type selected.
+     *
+     * @return a {@link KeyPair} with the key type selected
+     */
+    private KeyPair generateKeyPair() throws IllegalArgumentException {
+        final RadioButton rsaKeyButton = (RadioButton) findViewById(R.id.radio_rsa_key);
+        if (rsaKeyButton.isChecked()) {
+            return serviceLocator.getKeyGenerator().generateRSAKeyPair("NervousFish generated key");
+        }
+
+        throw new IllegalArgumentException("The selected key is not implemented");
+    }
+
+    /**
+     * Validates if the input fields are not empty and if the input is valid.
+     * This also means that the password and the repeat password should be the same,
+     * and the password length is larger or equal to 6.
+     *
+     * @return a {@link boolean} which is true if all fields are valid
+     */
+    private boolean validateInputFields() {
```

Comment:
Can you split this method up in smaller methods? BetterCodeHub doesn't like this length method